analyses/cell-type-ewings/cell_type_annotations/SCPCS000490/SCPCL000822_tumor-cell-validation.Rmd

```diff
@@ -0,0 +1,492 @@
+
+Next, we will plot the expression of each gene in each cell using a heatmap. 
+Here the rows are the marker genes and the columns are the cells with the color indicating the expression of that gene in that cell. 
+
+```{r}
+# create matrix with marker genes as rows and barcodes as columns
+marker_gene_heatmap <- coldata_df |> 
+  dplyr::select(gene_expression, gene_symbol, barcodes) |> 
+  tidyr::pivot_wider(values_from = gene_expression, 
+                     names_from = barcodes) |> 
+  tibble::column_to_rownames("gene_symbol") |> 
+  as.matrix()
+
+# create annotation df, make sure that dimensions match up by only getting unique barcodes 
+annotation_df <- coldata_df |> 
+  dplyr::select(barcodes, combined_cnv_classification) |> 
+  unique() 
+
+annotation <- ComplexHeatmap::columnAnnotation(
+  classification = annotation_df$combined_cnv_classification,
+  col = list(classification = c("Tumor" = "#00274C", "Normal" = "#FFCB05")))
+```
+
+
+```{r}
+# plot heatmap of marker genes 
+ComplexHeatmap::Heatmap(
+  marker_gene_heatmap,
+  col = circlize::colorRamp2(c(0, 1), colors = c("white", "#00274C")),
+  border = TRUE,
+  ## Row parameters
+  cluster_rows = TRUE,
+  row_title = "Marker gene symbol",
+  row_title_gp = grid::gpar(fontsize = 12),
+  row_title_side = "left",
+  row_names_side = "left",
+  row_dend_side = "right",
+  row_names_gp = grid::gpar(fontsize = 10),
+  ## Column parameters
+  cluster_columns = TRUE,
+  show_column_names = FALSE,
+  bottom_annotation = annotation
+)
+```
+As expected, the tumor cells appear to have more marker genes that are expressed than normal cells. 
+
+### Gene set scores
+
+Next, we will look at the gene set scores for three different EWS-FLI1 gene sets. 
+The below plots will all use the `mean-GENESET` columns. 
+
+First we will just look at the gene set scores for each cell on the UMAP. 
+
+```{r}
+# plot tumor cells for each classification method 
+geneset_plot_df <- coldata_df |> 
+  dplyr::select(barcodes, UMAP1, UMAP2, ends_with("classification"), starts_with("mean")) |> 
+  tidyr::pivot_longer(
+    cols = starts_with("mean"),
+    names_to = "geneset",
+    values_to = "mean_score"
+  ) |> 
+  dplyr::mutate(
+    geneset = stringr::word(geneset, -1, sep = "-")
+  )
+
+ggplot(geneset_plot_df, aes(x = UMAP1, UMAP2, color = mean_score)) +
+  geom_point(size = 0.5, alpha = 0.5) +
+  facet_wrap(vars(geneset)) +
+  scale_color_viridis_c()
+```
+
+Generally, the higher gene set scores appear to be in the group of cells that are tumor cells with some slight variations between gene sets. 
+
+Below we look at gene set scores for cells classified as tumor cells vs normal cells for all classification methods. 
+
+```{r, fig.height=10}
+geneset_plot_df |> 
+  tidyr::pivot_longer(
+    cols = ends_with("classification"),
+    names_to = "method",
+    values_to = "classification"
+  ) |> 
+  ggplot(aes(x = mean_score, color = classification)) +
+  geom_density(bounds = c(0,1), bw = 0.05) +
```

Comment:
I don't think you want bound here, as the data you are plotting is not constrained to 0,1. 
```suggestion
  geom_density(bw = 0.05) +
```